factories/solrSearcherFactory.js

```diff
@@ -171,6 +178,62 @@
       });
     }
 
+    function explainOther (otherQuery, fieldSpec) {
+      /*jslint validthis:true*/
+      var self = this;
+
+      var getOverridingExplain = function(doc, fieldSpec, overridingExplains) {
+        var idFieldName = fieldSpec.id;
+        var id = doc[idFieldName];
+
+        if (id && overridingExplains && overridingExplains.hasOwnProperty(id)) {
+          return overridingExplains[id];
+        }
+        return null;
+      };
+
+      var docsWithExplainOther = function(newDocs, fieldSpec, othersExplained) {
+        var docs = [];
+
+        angular.forEach(newDocs, function(doc) {
+          var overridingExplain = getOverridingExplain(doc, fieldSpec, othersExplained);
+          var normalDoc         = normalDocsSvc.createNormalDoc(fieldSpec, doc, overridingExplain);
```

Comment:
So this service has intentionally avoided depending on normalDocsSvc on purpose, instead producing regular ole Solr/ES docs and expecting the user to normalize them. I would prefer not to have this dependency in the search services, but I need to dig more into what you're doing.